src/System.Management.Automation/FormatAndOutput/common/StringDecorated.cs

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Text.RegularExpressions;
+
+namespace System.Management.Automation.Internal
+{
+    /// <summary>
+    /// Extensions to String type to calculate and render decorated content.
+    /// </summary>
+    public class StringDecorated
+    {
+        private const char ESC = '\x1b';
+        private readonly bool _isDecorated;
+        private readonly string _text;
+        private string _plaintextcontent;
+
+        private string _plaintext
+        {
+            get
+            {
+                if (_plaintextcontent == null)
+                {
+                    _plaintextcontent = _ansiRegex.Replace(_text, string.Empty);
+                }
+
+                return _plaintextcontent;
+            }
+        }
+
+        // replace regex with .NET 6 API once available
+        private static readonly Regex _ansiRegex = new Regex(@"\x1B(?:[@-Z\\-_]|\[[0-?]*[ -/]*[@-~])", RegexOptions.Compiled);
+
+        /// <summary>
+        /// Initializes a new instance of the StringDecorated class.
+        /// </summary>
+        /// <param name="text">The input string.</param>
+        public StringDecorated(string text)
+        {
+            _text = text;
+            _isDecorated = text.Contains(ESC);
+        }
+
+        /// <summary>
+        /// Gets a value indicating whether the string contains decoration.
+        /// </summary>
+        /// <returns>Boolean if the string contains decoration.</returns>
+        public bool IsDecorated => _isDecorated;
+
+        /// <summary>
+        /// Gets the length of content sans escape sequences.
+        /// </summary>
+        /// <returns>Length of content sans escape sequences.</returns>
+        public int ContentLength => _plaintext.Length;
+
+        /// <summary>
+        /// Return a substring based on content length starting from index 0.
+        /// </summary>
+        /// <param name="length">Number of characters from content to return.</param>
+        /// <returns>A string up to the number of specified characters including decoration up to that length.</returns>
+        public string Substring(int length)
+        {
+            return _text;
```

Comment:
Seems you are missing something here.

Reply:
You're right, this was a placeholder and I didn't get to completing it.  I'll remove it for now since it's not being used.  Ideally would like .NET API support rather than doing something custom for this.